Mathlib/Analysis/SpecialFunctions/BinaryEntropy.lean

```diff
@@ -0,0 +1,428 @@
+/-
+Copyright (c) 2023 Adomas Baliuka. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adomas Baliuka
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.NegMulLog
+import Mathlib.Analysis.Convex.SpecificFunctions.Basic
+
+/-!
+# Properties of Shannon q-ary entropy and binary entropy functions
+
+The [binary entropy function](https://en.wikipedia.org/wiki/Binary_entropy_function)
+`binEntropy p := - p * log p - (1 - p) * log (1 - p)`
+is the Shannon entropy of a Bernoulli random variable with success probability `p`.
+
+More generally, the q-ary entropy function is the Shannon entropy of the random variable
+with possible outcomes `{1, ..., q}`, where outcome `1` has probability `1 - p`
+and all other outcomes are equally likely.
+
+`qaryEntropy (q : ℕ) (p : ℝ) := p * log (q - 1) - p * log p - (1 - p) * log (1 - p)`
+
+This file assumes that entropy is measured in Nats, hence the use of natural logarithms.
+Most lemmas are also valid using a logarithm in a different base.
+
```

Comment:
Please add a list with the main declarations. Are there some lemmas in this file that you think deserve to be listed at the top of the file? If so, please add them to the list.
```suggestion

## Main declarations

* `Real.binEntropy`: the binary entropy function
* `Real.qaryEntropy`: the `q`-ary entropy function

```

Reply:
I think there's no need to add lemmas there. `NegMulLog.lean` also does not add any. We could copy the definitions for the two functions there.

Reply:
I think concavity could go in there, maybe also monotonicity.

Reply:
I wrote something

Reply:
Perfect, thanks!